toolkit/resources/imageconfigs/additionalfiles/iso_initrd/root/runliveinstaller

```diff
@@ -83,6 +84,38 @@ else
     retry "{ mount -L $LABEL -o ro $ISO_ROOT || grep -qs $ISO_ROOT /proc/mounts ; }"
 fi
 
+# If we are running unattended (i.e., have $UNATTENDED_CONFIG_FILE), we need to make sure that we only run once.
+# We can use flock to synchronize the unattended install flows.
+if [[ -f "$UNATTENDED_CONFIG_FILE" ]]; then
+    echo "Synchronizing unattended install flows... current sessions are:"
+    who
+    thisTerminal=$(who -m | xargs | cut -d' ' -f2)
+    echo "This instance is on '$thisTerminal'"
+
+    # Generally we want to prioritize ttyS0 since that will be used for debugging. The output from the log file will
+    # contain the timestamp and color codes so it is still useful for debugging and showing liveness but it will look
+    # messy.
+    if [[ "$thisTerminal" != "ttyS0" ]]; then
```

Comment:
If we're sure that `ttyS0` is doing the same thing, why not just exit here immediately?

Reply:
I think it's expected, since `runliveinstaller` is the entrypoint script for each user session: https://github.com/microsoft/CBL-Mariner/blob/2.0/toolkit/resources/imageconfigs/iso_initrd.json#L50

Reply:
I never want to leave us in the state where the installer doesn't run at all. I don't know if we can guarantee the exact consoles we have won't ever change. This is much more defensive than explicitly bailing out on a specific console.